relentless/simulate/hoomd.py

```diff
@@ -304,24 +312,34 @@ def __call__(self, sim):
         ------
         ValueError
             If the simulation box dimensions specified by the file is inconsistent
-            with the ensemble volume (for a constant volume simulation).
+            with the ensemble extent (for a constant extent simulation).
 
         """
         with sim.context:
             sim.system = hoomd.init.read_gsd(self.filename,**self.options)
 
-            # check that the boxes are consistent in constant volume sims.
+            # check that the boxes are consistent in constant extent sims.
             if sim.ensemble.V is not None:
                 system_box = sim.system.box
                 box_from_file = numpy.array([system_box.Lx,
                                           system_box.Ly,
                                           system_box.Lz,
                                           system_box.xy,
                                           system_box.xy,
-                                          system_box.yz])
+                                          system_box.yz,
+                                          system_box.dimensions])
                 box_from_ensemble = self.extract_box_params(sim)
                 if not numpy.all(numpy.isclose(box_from_file,box_from_ensemble)):
-                    raise ValueError('Box from file is is inconsistent with ensemble volume.')
+                    raise ValueError('Box from file is is inconsistent with ensemble extent.')
+                
+                dim_from_file = system_box.dimensions
+                if isinstance(sim.ensemble.V, Area):
+                    dim_from_ensemble = 2
+                elif isinstance(sim.ensemble.V, Volume):
+                    dim_from_ensemble = 3
+                if not numpy.isclose(dim_from_file,dim_from_ensemble):
+                    raise ValueError('Dimensions from file is is inconsistent with ensemble dimensions.')
```

Comment:
You could remove this if you make the fix above, the dimension should be covered by the return value of the extract function.